plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -311,4 +344,29 @@ private static boolean containSameElements(Iterable<? extends ColumnHandle> firs
     {
         return ImmutableSet.copyOf(first).equals(ImmutableSet.copyOf(second));
     }
+
+    private enum SystemTableHandler
```

Comment:
This abstraction seems unnecessary. There's only one type of handler, and it's not clear that future ones would use the same suffix matching logic. If and when we need to add others, we can refactor the code to use a generic enum if appropriate.